automated-release/github-release/index.js

```diff
@@ -0,0 +1,71 @@
+const GithubApi = require('github-api')
+const request = require('request-promise-native')
+const fs = require('fs')
+
+module.exports = class GithubRelease {
+
+    constructor(token) {
+        this.github = new GithubApi({ token: token })
+        this.token = token
+    }
+
+    async performRelease(repoOptions, prOptions, releaseOptions, artifacts) {
+        const repo = this.github.getRepo(repoOptions.owner, repoOptions.name)
+        const pullRequest = await this._createReleasePullRequest(repo, prOptions)
+        const release = await this._createRelease(repo, releaseOptions)
+        await this._uploadAssets(repoOptions, release.id, artifacts)
+        await this._commentOnPr(repoOptions, pullRequest.number, release.name, release.html_url)
+    }
+
+    _createReleasePullRequest(repo, request) {
+        const createOptions = {
+            title: request.title,
+            body: request.body,
+            head: request.fromBranch,
+            base: request.intoBranch
+        }
+        return repo.createPullRequest(createOptions).then(result => result.data)
+    }
+
+    _createRelease(repo, release) {
+        const releaseOptions = {
+            tag_name: release.tag,
+            target_commitish: release.target,
+            name: release.title,
+            body: release.body,
+            draft: release.isDraft,
+            prerelease: release.isPreRelease
+        }
+        return repo.createRelease(releaseOptions).then(result => result.data)
+    }
+
+    async _uploadAssets(repoOptions, releaseId, artifacts) {
+        const allArtifacts = artifacts.map((artifact) => this._uploadAsset(repoOptions, releaseId, artifact))
```

Comment:
you mean without the single param with `(*)`? I've started always adding them because `typescript` forces it